ocean_provider/validation/requests.py

```diff
@@ -48,6 +68,103 @@ def __init__(
         self._processor = CustomRulesProcessor()
 
 
+class RBACValidator:
+    def __init__(self, request_name=None, request=None, assets: Optional[list] = None):
+        self._request = request
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._assets = assets
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @property
+    def component(self):
+        return self._component
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def assets(self):
+        return self._assets
+
+    @property
+    def request(self):
+        return self._request
+
+    def passes(self):
+        return True
+
+    def fails(self):
+        return False
+
+    def build_payload(self):
+        payload = {
+            "eventType": self.action,
+            "component": self.component,
+            "providerAddress": get_provider_wallet().address,
+            "credentials": self.credentials,
+        }
+        payload.update(getattr(self, f"build_{self.action}_payload")())
+        return payload
+
+    def build_encryptUrl_payload(self):
+        message = "encryptUrl" + json.dumps(self.credentials)
+        signature = sign_hash(
```

Comment:
I think there's a signing function somewhere. If not, you can create it. Can we also add the provider wallet to the list of properties, to avoid using the function every time?